zookeeper-server/src/main/java/org/apache/zookeeper/server/NettyServerCnxn.java

```diff
@@ -520,6 +520,9 @@ private void receiveMessage(ByteBuf message) {
                         if (len < 0 || len > BinaryInputArchive.maxBuffer) {
                             throw new IOException("Len error " + len);
                         }
+                        if (zkServer == null || !zkServer.isRunning()) {
+                            throw new IOException("ZK down");
+                        }
```

Comment:
sorry I missed in previous review, I think it is better to handle the same way as it is handled in `NIOServerCnxn` to be consistent with all types of connections
```
if (!isZKServerRunning()) {
      throw new IOException("ZooKeeperServer not running");
  }
```

Reply:
actually I would prefer to use the following approach, copied from an other part of this function: https://github.com/apache/zookeeper/blob/7fdadf7273f34dd0552db25a3771cf55b65e9208/zookeeper-server/src/main/java/org/apache/zookeeper/server/NettyServerCnxn.java#L475-L478

This way we kind of "save" the reference of `zkServer` to an other variable and if `zkServer` becomes null right after we did the check, we are still in a safe spot. Although the final solution would be to use locks and not allowing to use/modify `zkServer` paralelly from multiple threads... But that would be a more serious refactoring.

Reply:
I just pushed a new commit, what do you think?

Reply:
Great!